OriginProtocol/origin
# Natasha/photo upload orientation

### Description:

[Issue #828](https://github.com/OriginProtocol/origin/issues/828)

- Currently when a user uploads a photo, we are not managing orientation (this is a browser-wide issue).

**This PR:**
- implements [Javascript-Load-Image](https://github.com/blueimp/JavaScript-Load-Image) to deal with image orientation, cropping & sizing.
- changes the behavior of the `generateCroppedImage` method to use Javascript-Load-Image
- Fix css issue on mobile for Edit Profile modal photo area
- Extracts the basic localStorage wrapper in the Onboarding/reducer (used in an early iteration of fixing photo orientation)

### Manual Testing:
- Upload a photo in the following scenarios (on a physical device -- especially iPhone):
  1. In a conversation
     - an image wider/larger than 1000px should be resized
  2. When creating a Listing
    - initial upload should attempt to auto-crop center
    - 4/3 aspect ratio
  3. When adding/updating Profile photo
    - aspect ratio of 1

- The orientation & cropping should be correct. Orientation should be tested on iPhone especially (see issue linked above).
- This PR should be tested in different browsers
### Screenshots
**Profile Photo**

![image](https://user-images.githubusercontent.com/17957496/49182051-6ab6e000-f316-11e8-868a-6c0b276f961c.png)

**New Listing**

![image](https://user-images.githubusercontent.com/17957496/49182115-95089d80-f316-11e8-9eea-2a528c76157b.png)

**Conversation**
![image](https://user-images.githubusercontent.com/17957496/49182143-ace02180-f316-11e8-8b41-0291113e1259.png)

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] Wrap any new text/strings for translation
- [ ] Map any new environment variables with a default value in the Webpack config
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)
